lightning/src/ln/channel.rs

```diff
@@ -2761,6 +2765,12 @@ impl<Signer: Sign> Channel<Signer> {
 			*self.next_remote_commitment_tx_fee_info_cached.lock().unwrap() = None;
 		}
 
+		let secret = SecretKey::from_slice(&msg.per_commitment_secret)
```

Comment:
Why eat the cost of converting to a `SecretKey` and validating it when we could just pass it as a `[u8; 32]` which it already is and presumably will be validated as (as its a SHA-256 operation on a secret)?

Reply:
The Signer wants the `SecretKey` because it needs to further convert to a point to make sure it's the right point.

Actually, we are already calculating the secret earlier in the function, for obvious reasons.  Will DRY this.